src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -223,6 +223,9 @@ Copyright (c) .NET Foundation. All rights reserved.
       <_ReadyToRunOutputPath>$(IntermediateOutputPath)R2R</_ReadyToRunOutputPath>
     </PropertyGroup>
 
+    <!-- TODO: Temp error to limit R2R compilation to self-contained mode only. Delete this when fixing #3109 and #3110 -->
+    <NETSdkError Condition="'$(SelfContained)' != 'true'" ResourceName="ReadyToRunTargetNotSuppotedError" />
```

Comment:
Should this check be upfront (say in prepare for publish) before we do any work for R2R compilation? 
Not a significant issue, since this is only an interim check.

Reply:
It is upfront. It runs before the PrepareForReadyToRunCompilation task, which runs before any compilation